src/app/api/openai/chat/route.ts

```diff
@@ -1,19 +1,28 @@
+import { NextAuthRequest } from 'next-auth/lib';
+
 import { OpenAIChatStreamPayload } from '@/types/openai/chat';
 
 import { getPreferredRegion } from '../../config';
+import { auth } from '../../next-auth';
 import { createBizOpenAI } from '../createBizOpenAI';
 import { createChatCompletion } from './createChatCompletion';
 
 export const runtime = 'edge';
 export const preferredRegion = getPreferredRegion();
 
-export const POST = async (req: Request) => {
-  const payload = (await req.json()) as OpenAIChatStreamPayload;
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const POST = auth(
+  async (
+    req: NextAuthRequest,
+    { params: _params }: { params: Record<string, string | string[] | undefined> },
+  ) => {
+    const payload = (await req.json()) as OpenAIChatStreamPayload;
 
-  const openaiOrErrResponse = createBizOpenAI(req, payload.model);
+    const openaiOrErrResponse = createBizOpenAI(req, payload.model);
```

Comment:
I think maybe it's better to refactor the authentication from the `createBizOpenAI` due to there are several ways to verify. like the `auth` wrapper function.